backend/src/controller/listing.ts

```diff
@@ -35,14 +42,106 @@ module.exports.getListing = (req: Request, res: Response) => {
 };
 
 module.exports.getFullInfoListings = (req: Request, res: Response) => {
+  fillFullInfo();
   res.json(fullInfoListings);
 };
 
 module.exports.getFullInfoListing = (req: Request, res: Response) => {
+  fillFullInfo();
   const fullInfoListing = fullInfoListings.find(
     (Listing) => Listing.listingID === req.params.id
   );
   fullInfoListing
     ? res.json(fullInfoListing)
-    : res.send("listing doesn`t exist");
+    : res.status(404).json({ errorMessage: "Rosources not found" });
+};
+
+module.exports.addListing = (req: Request, res: Response) => {
+  if (typeof req.body == undefined) {
+    res.json({
```

Comment:
suggestion: set status 400 (bad request) `res.status(400).json({// rest of the reponse})`